onebusaway-android/src/amazon/java/org/onebusaway/android/map/googlemapsv2/BaseMapFragment.java

```diff
@@ -1009,6 +1036,42 @@ public void onClick(DialogInterface dialog, int which) {
                             new DialogInterface.OnClickListener() {
                                 @Override
                                 public void onClick(DialogInterface dialog, int which) {
+                                    createZipCodeDialog().show();
+                                }
+                            }
+                    );
+            return builder.create();
+        }
+
+        private Dialog createZipCodeDialog() {
+            Drawable icon = getResources().getDrawable(android.R.drawable.ic_dialog_map);
+            DrawableCompat.setTint(icon, getResources().getColor(R.color.theme_primary));
+            final EditText zipInput = new EditText(this.getContext());
+            zipInput.setInputType(InputType.TYPE_CLASS_NUMBER);
+
+            final AlertDialog.Builder builder = new AlertDialog.Builder(getActivity())
+                    .setTitle(R.string.main_ziplocation_title)
+                    .setIcon(icon)
+                    .setCancelable(false)
+                    .setMessage(R.string.main_ziplocation)
+                    .setView(zipInput)
```

Comment:
This needs margins on the sides of the EditText, so it aligns with the rest of the dialog content and meets the material design guidelines:

![image](https://cloud.githubusercontent.com/assets/928045/16155232/66c496b4-347c-11e6-9710-d40465f48141.png)

You may be able to do this programmatically, but it's probably easier just to use a layout.